platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Utilities.java

```diff
@@ -133,45 +135,43 @@ public static int arrayCompare(byte[] sig1, byte[] sig2) {
      * XORed with whitening before the comparison. The XOR doesn't actually happen, and the arrays are left
      * unchanged.
      *
-     * @param sig1
+     * @param a1
      * 		first array
-     * @param sig2
+     * @param a2
      * 		second array
      * @param whitening
      * 		the array virtually XORed with the other two
      * @return 1 if first is bigger, -1 if second, 0 otherwise
      */
-    public static int arrayCompare(byte[] sig1, byte[] sig2, byte[] whitening) {
-        int maxLen;
-        int minLen;
-        if (sig1 == null && sig2 == null) {
+    public static int arrayCompare(@Nullable final Bytes a1, @Nullable final Bytes a2, byte[] whitening) {
+        if (a1 == null && a2 == null) {
             return 0;
         }
-        if (sig1 != null && sig2 == null) {
+        if (a1 != null && a2 == null) {
             return 1;
         }
-        if (sig1 == null && sig2 != null) {
+        if (a1 == null && a2 != null) {
             return -1;
         }
-        maxLen = Math.max(sig1.length, sig2.length);
-        minLen = Math.min(sig1.length, sig2.length);
+        final int maxLen = (int) Math.max(a1.length(), a2.length());
```

Comment:
I just ask myself if an if might be better for performance & memory since we do not need to downcast long to int. Never proofed it.

Reply:
I think that would be considered premature optimization